src/InfluxDB2/UdpWriter.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+
+namespace InfluxDB2;
+
+/**
+ * Class UdpWriter
+ * @package InfluxDB2
+ *
+ * UDP Writer Requirements:
+ * 1. Installed ext-sockets
+ * 2. Since Influxdb 2.0+ does not support UDP protocol natively you need to install and configure Telegraf plugin:
+ *    https://docs.influxdata.com/telegraf/v1.16/plugins/#socket_listener
+ * 3. Extra config option passed to client: udpPort. Optionally you can specify udpHost, otherwise udpHost will parsed from url option
+ *
+ * Example:
+ * $client = new InfluxDB2\Client(["url" => "http://localhost:8086", "token" => "my-token",
+ *       "bucket" => "my-bucket",
+ *       "org" => "my-org",
+ *       "precision" => InfluxDB2\Model\WritePrecision::NS,
+ *       "udpPort" => 8094,
+ *   ]);
+ *   $writer = $client->createUdpWriter();
+ *   // Write parameter matches WriterApi, so you can write strings, Point objects, arrays
+ *   $writer->write('h2o,location=west value=33i 15');
+ */
+class UdpWriter implements Writer
+{
+
+    public $options = [];
+
+    /**
+     * @var resource
+     */
+    protected $socket;
+
+    /**
+     * UdpWriter constructor.
+     * @param array $options
+     * @throws \Exception
+     */
+    public function __construct($options)
+    {
+        $this->options = $options;
+        if (empty($this->options['udpPort'])) {
+            throw new \Exception('udpPort option does not specified');
+        }
+        if (empty($this->options['udpHost'])) {
+            $this->options['udpHost'] = parse_url($this->options['url'], PHP_URL_HOST);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function write($data)
+    {
+        $payload = null;
+        if (is_string($data)) {
+            $payload = $data;
+        }
+        if ($data instanceof Point) {
+            $payload = $data->toLineProtocol();
+        }
+        if (is_array($data)) {
+            if (!array_key_exists('name', $data)) {
+                foreach ($data as $item) {
+                    if (isset($item)) {
+                        $this->write($item);
+                    }
+                }
+                return;
+            }
+            $payload = Point::fromArray($data)->toLineProtocol();
+        }
```

Comment:
Please, reuse serialization from https://github.com/influxdata/influxdb-client-php/blob/fa14d43d5fe0e428438f670f6480fc27783a4f8a/src/InfluxDB2/WriteApi.php#L212

Something like following code will be easier to maintenance:

```php
<?php

namespace InfluxDB2;

class WritePayloadSerializer
{
    /**
     * Generate payload from provided data.
     *
     * @param $data string|Point|array to generate payload
     * @param string|null $precision the precision used as a key for Batch
     * @param string|null $bucket the bucket used as a key for Batch
     * @param string|null $org the org used as a key for Batch
     * @param WriteType|null $writeType specify type of writes - Synchronous or Batching
     * @return BatchItem|string|null
     */
    public static function generatePayload($data, string $precision = null, string $bucket = null, string $org = null, WriteType $writeType = null)
    {
        if ($data == null || empty($data)) {
            return null;
        }
        if (is_string($data)) {

            if (WriteType::BATCHING == $writeType) {
                return new BatchItem(new BatchItemKey($bucket, $org, $precision), $data);
            } else {
                return $data;
            }
        }
        if ($data instanceof Point) {
            return self::generatePayload($data->toLineProtocol(), $data->getPrecision() !== null ?
                $data->getPrecision() : $precision, $bucket, $org);
        }
        if (is_array($data)) {
            if (array_key_exists('name', $data)) {
                return self::generatePayload(Point::fromArray($data), $precision, $bucket, $org);
            }

            $payload = '';

            foreach ($data as $item) {
                if (isset($item)) {
                    $payload .= self::generatePayload($item, $precision, $bucket, $org) . "\n";
                }
            }

            // remove last new line
            if (isset($payload) && trim($payload) !== '') {
                $payload = rtrim($payload, "\n");
            }

            return $payload;
        }

        return null;
    }
}
```

```suggestion
        $payload = WritePayloadSerializer::generatePayload($data);
```